docs/en/advanced_guides/detection_transformer.md

```diff
@@ -0,0 +1,230 @@
+# Detection Transformer
+
+Detection Transformers (DETRs) are a series of Transformer-based object detectors.
+
+In most DETRs, the features extracted from backbone and neck were fed into a Transformer which is composed of an encoder and a decoder. The Transformer directly outputs a set of queries in parallel. Each query corresponds to one prediction, which may be an object or `no object`.
+
+![DETR_overall](.\DETR_overall.png)
+
+There is a brand new implementation of DETRs in MMDetection, whose features are fourfold:
+
+#### 1. A unified base detector class for DETRs.
+
+According to the existing DETRs, a unified base detector class `DetectionTransformer` is designed to standardize the architectures and forward procedures of DETRs. (mmdet/models/detectors/base_detr.py)
+
+Each detector inherited from `DetectionTransformer` is mainly consists of six components: `backbone`, `neck`, `positional_encoding`, `encoder`, `decoder`, and `bbox_head`. The `backbone` and `neck` extract features of input batch. The `positional_encoding` encodes the position of each feature points. The `encoder` processes output features of neck. The `decoder` pools encoder features of objects into the queries. At last, the `bbox_head` makes predictions and calculates losses on output queries of decoder.
+
+For the forward process, except for the earliest `extracting features` and the latest `calculating with head`, the intermediate forward process of transformer are designed as four steps: `pre_transformer`, `forward_encoder`, `pre_decoder`, and `forward_decoder`. The parameters flow among the functions are summarized as follow:
+
+![DETR_forward_process](.\DETR_forward_process.png)
+
+The corresponding implementation is as follow:
+
+```Python
+def forward_transformer(self, img_feats: Tuple[Tensor],
+                        batch_data_samples: OptSampleList = None) -> Dict:
+
+    encoder_inputs_dict, decoder_inputs_dict = self.pre_transformer(
+        img_feats, batch_data_samples)
+
+    encoder_outputs_dict = self.forward_encoder(**encoder_inputs_dict)
+
+    tmp_dec_in, head_inputs_dict = self.pre_decoder(**encoder_outputs_dict)
+    decoder_inputs_dict.update(tmp_dec_in)
+
+    decoder_outputs_dict = self.forward_decoder(**decoder_inputs_dict)
+    head_inputs_dict.update(decoder_outputs_dict)
+
+    return head_inputs_dict
+```
+
+#### 2. More concise components.
+
+The implementations of most components, such as `DetrTransformerEncoderLayer`, `DetrTransformerDecoder`, and `DetrHead`, become more concise.
+
+The `XTransformer` class has been dropped. Instead, the detector initialize `XTransformerEncoder` and `XTransformerDecoder` directly.
+
+The original intricate modules of Transformer, including `XTransformerEncoder`, `XTransformerEncoderLayer`, `XTransdormerDecoder`, `XTransformerDecoderLayer`, are replaced with straightforward modules which attend to clear logic rather than excessive compatibility. For example:
```

Comment:
Should we add the original implementation version of `DetrTransformerEncoder` for better comparison?

Reply:
@ZCMax  Is there any manner of folding a code block in markdown? I tried html5, but the effect is bad. UGLY
Both the original and new implementations are long, i think making them fold will be better.